addons/account_edi/models/account_edi_format.py

```diff
@@ -406,15 +404,45 @@ def _update_invoice_from_attachment(self, attachment, invoice):
                     return res
         return self.env['account.move']
 
+    ####################################################
+    # Helpers
+    ####################################################
+
+    def _is_edi_instance_of(self, edi_code):
+        ''' Helper to check if an EDI is using the current code passed as parameter or is inhering one using
+        the 'parent_id' field.
+
+        :param edi_code:    An account.edi.format code.
+        :return:            True if 'edi_code' is part of this format, False otherwise.
+        '''
+        self.ensure_one()
+        codes = set()
+        edi_format = self
+        while edi_format:
+            codes.add(edi_format.code)
+            edi_format = edi_format.parent_id
+        return edi_code in codes
+
     ####################################################
     # Import helpers
     ####################################################
+    def _get_namespaces(self, tree):
+        ''' If the namespace is declared with xmlns='...', the namespaces map contains the 'None' key that causes an
+        TypeError: empty namespace prefix is not supported in XPath
+        Then, we need to remap arbitrarily this key.
```

Comment:
I agree this is fuzzy... This is actually the case for every xml where there is a namespace with empty prefix (xmlns=something), which is often the case (not only ubl). The namespace manipulation in etree is a pain, for `find´, `find_all` and `xpath` you need to give the namespaces. lxml eased things a bit by providing an easy way to retrieve the namespaces by using `root.nsmap`, however this returns also the namespaces with empty prefix, which are not supported by these methods (generates an error). Hence the need to remap it.

You should do that every time you want to use theses methods, which in the EDI happens very often. Previously, this method was defined as an inner method of a lot of import methods. Rather than defining multiple times, I think it's good to have this helper that is defined once and can be used everywhere. In the future I would like to add some more helpers for the more common stuff that we need to do in the edi (finding values mostly). 

Again, I agree, this is a pain that we have to do that this way. If you have a better suggestion, I'm open.

Reply:
The reasons seem legit, but the docstring is difficult to understand.  So we filter empty from root.nsmap?  (would not lxml have a helper method itself?)

Reply:
The explanation you gave here is a lot clearer.

Reply:
Use `find({*}<my_tag>)`

Reply:
Instead we skip the namespaces using `element.find('./{*}myTag')`